### PR TITLE
[READY]Staff of the Locker

### PR DESCRIPTION
**Pull request by Time-Green**
Adds a magical locker staff. 
It goes through people, capturing anyone being hit by it and putting them in a welded locker
https://gyazo.com/39b4c456bdca504eb85df3493847f67e

The locker decays in about 5 minutes, to prevent the hallways flooding too much.
Escape time is only a minute.
Explosions are less effective on people inside
:cl: Time-Green and locker sprites by MrDoomBringer
add: Reports have come in that the wizard federation has harnessed some of the ancient locker force to create a wand
/:cl:
Why: It's a bulky wand, and great if you want people to just fuck off. You can either leave them or do something horrible to them.

The sprites are codersprites, except for the locker, please forgive me

### CODE REVIEW

**Comment by Borisvanmemes** [comment]
I fucking love it, TEN OUTTA TEN PR!

File: code/modules/projectiles/projectile/magic.dm [hunk 1]
```diff
@@ -335,6 +335,57 @@
 			return
 	. = ..()
 
+
+/obj/item/projectile/magic/locker
+	name = "locker bolt"
+	icon_state = "locker"
+	nodamage = 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/projectile/magic.dm]: boo?

**Comment by Nickvr628** [comment]
>uses staff of locker on crew
>uses staff of animation on the locker

speedmerge pls

**Comment by Time-Green** [comment]
requires two staffs though, but yeah

File: code/modules/projectiles/projectile/magic.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/projectiles/projectile/magic.dm]: as in boolean

**Comment by PKPenguin321** [comment]
A "magic locker" with a shorter breakout time for the weld might be less annoying

**Comment by Nabski89** [comment]
Can you describe what you'd like the actual staff to look like?

I'm assuming "locker on a stick" is NOT going to be the final answer here.

Do you also need a magical locker sprite?

I suppose this could also be done with not lockers but instead putting people in blocks of ice or crystal, but that's not as much of a meme as the lockstick.

**Comment by Time-Green** [comment]
The locker on a stick was the best I could do, but yeah it shouldn’t stay.
I don’t know a better design, I am very lacking in the sprite and visual department. 
Do you have something cool in mind? 

Also about the locker sprite, I should probably lower the escape time, and a special magical locker sprite would be great if you feel like it

Thanks alot

File: code/modules/projectiles/projectile/magic.dm (hunk 1 above)
Comment (Time-Green) [reply code/modules/projectiles/projectile/magic.dm]: right i forgot

**Comment by JJRcop** [comment]
Next magical tool: The Hotel Staff

**Comment by MrDoomBringer** [comment]
hello yes i made a quick magic locker sprite waddyall think
bad locker2

**Comment by Nabski89** [comment]
Can I at least animate it?

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 2]
```diff
@@ -316,6 +316,12 @@
 	cost = 1
 	category = "Defensive"
 
+/datum/spellbook_entry/item/lockerstaff
+	name = "Staff of the Locker"
+	desc = "A staff that shoots lockers. It eats anyone it hits on its way, leaving a welded locker with your victims behind. You can walk into the same direction as you shoot to put yourself into the locer for a quick travel, this wont weld the locker shut."
```
Comment (ChangelingRain) [inline code/modules/antagonists/wizard/equipment/spellbook.dm]: grammar!

File: code/modules/projectiles/guns/magic/staff.dm [hunk 3]
```diff
@@ -94,3 +94,15 @@
 	if(attack_type == PROJECTILE_ATTACK)
 		final_block_chance = 0
 	return ..()
+
+/obj/item/gun/magic/staff/locker
+	name = "staff of the locker"
+	desc = "An artefact that expells encapsulating bolts, for incapacitating thy enemy."
```
Comment (ChangelingRain) [inline code/modules/projectiles/guns/magic/staff.dm]: grammar!

**Comment by Iamgoofball** [comment]
is this carried by the Lords of the Locker Room?

**Comment by MrDoomBringer** [comment]
@Iamgoofball it is carried by Jim Locker.

**Comment by WJohn** [comment]
On the positive, this can be resisted out of and done so in less time than being turned into a statue, and friends can also help you out of it by unwelding the locker open. The staff can also be disarmed and used against the wizard so he can cut that shit out.

On the negative, this can be fired in rapid succession, is ranged, recharges relatively quickly, can capture multiple people per shot, and is still a type of player removal spell, even if temporary like the statue.

This definitely needs some playtesting to see how powerful (and annoying) this is. I'm also curious to see if the flying locker swallows up or blocks projectiles, as it could be a powerful safeguard against them by just running away while spamming it and being immune to all manner of ranged stuns trying to hit you (possibly catching your pursuer, too).

**Comment by Time-Green** [comment]
I am absolutely in love with the locker sprite
If the staff sprite stays, I’ll probably still need a sensible in-hand, I just resprited the broom because my first attempt was absolutely disgusting

**Comment by AnturK** [comment]
Make it override content explosion damage or this will turn into free kill with fireball.

File: code/modules/projectiles/projectile/magic.dm [hunk 4]
```diff
@@ -335,6 +335,57 @@
...
+
+/obj/item/projectile/magic/locker
+	name = "locker bolt"
+	icon_state = "locker"
+	nodamage = TRUE
+	flag = "magic"
+	var/weld = TRUE
+	var/created = FALSE //prevents creation of more then one locker if it has multiple hits
+
+/obj/item/projectile/magic/locker/Bump(atom/A)
```
Comment (AnturK) [inline code/modules/projectiles/projectile/magic.dm]: Projectiles have specific function for before hit functionality. That seems better place to put this.

**Review by AnturK — CHANGES_REQUESTED** [review]
There's some random pngs in the diff.

File: code/modules/projectiles/projectile/magic.dm (hunk 4 above)
Comment (Time-Green) [reply code/modules/projectiles/projectile/magic.dm]: I completely forgot about the pngs.
I’ll track down whatever proc it is

**Comment by Kagus** [comment]
Curse of the Warlock(er)...

Does the projectile get blocked by shields?

**Comment by Time-Green** [comment]
Nope, but it’s a really big projectile so dodging should be a little easier

**Comment by Kagus** [comment]
I was more just wondering if the projectile might get deleted by blocking, and then having Weirdness(tm) raise its ugly head if the dead projectile then tried to locker someone. I don't think anyone's worried about something _unbalanced_ getting added to wizards.

So since this thing counts as welded, it's not gonna get popped open by using Knock afterwards, right? Kind of a shame, that'd be a great "Just a prank, muggle!" build.

I forget how immovable rods work with lockers. It's probably fine.

**Comment by obscolene** [comment]
>stand outside chapel
>locker anyone who walks by
>drag them in and cremate

**Comment by AlexTheSysop42** [comment]
\>stand outside chapel
\>MM anyone who walks by 
\>drag them in and cremate

**Comment by Doppeltongue** [comment]
Magic lockers should Indiana Jones their occupants. Scientists should take longer to break out of them because they're neerrrrds.

**Comment by Ratvar** [comment]
You can cremate whole lockers.

**Comment by TheWiznard** [comment]
sssss

not better than whats here but what if it was more like this?

**Comment by ExcessiveUseOfCobblestone** [comment]
If you're using Doombringer's locker you should make the wand look similar.

**Comment by Time-Green** [comment]
Dont need to change projectiles and the wand to the green version, I’ve got a pretty cool looking alternative worked out

**Comment by Time-Green** [comment]
I added some stuff, lockers now look cool thanks to mrdooms sprites.
It did a cool phase into the magical sprite, but apparently you cant animate overlays and ive been trying for so fucking long ive gotten absolutely fucking sick and tired of it.
For anyone wondering, this used to be the code
   var/image/I = image(src.icon,"[magic_icon]_whole",layer = src.layer+1)
    I.alpha = 0
    add_overlay(I)
    animate(I, alpha = 255, time = 20)
Note that it did not animate at all, and the animate proc might aswell just not have been there

**Comment by optimumtact** [comment]
this is the ugliest looking shit I have ever seen :-1:

**Comment by lolman360** [comment]
"The sprites are codersprites, please forgive me"

**Comment by JJRcop** [comment]
@MrDoomBringer's sprites for the locker are missing, is that intentional?

**Comment by Kagus** [comment]
> this is the ugliest looking shit I have ever seen 👎

Says the man who added khakis to the secdrobe.

**Comment by optimumtact** [comment]
life aint fair

**Comment by ExcessiveUseOfCobblestone** [comment]
murderbone wiz is okay unless it's lower pop wiz.

friendly wiz is never okay.

do not friendly antag

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 5]
```diff
@@ -265,6 +267,9 @@
 	else
 		return FALSE
 
+/obj/structure/closet/proc/weld_act(weld_state)
```
Comment (ShizCalev) [inline code/game/objects/structures/crates_lockers/closets.dm]: mind naming this something else like `after_weld` or something so someone doesn't read this and confuse it with `welder_act`?

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Pull request merged**